File: ruby/consecutive_substrings.rb
def consecutive_substrings(string)
  # type your code in here
  collector = []
  start = 0
  num_of_chars = 1
  while (num_of_chars <= string.length) do
    collector << string[start, num_of_chars]
    if num_of_chars == string.length 
      string.slice!(0)
      num_of_chars = 0
    end
    num_of_chars += 1
  end
  collector
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: ['a', 'ab', 'abc', 'b', 'bc', 'c']"
  print "=> " 
  print consecutive_substrings('abc')

  puts

  puts "Expecting: ['a']"
  print "=> " 
  print consecutive_substrings('a')

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution

File: javascript/consecutive_substrings.js
function consecutiveSubstrings(string) {
  const collector = []
  const start = 0
  let end = 1
  let newString = string
  while (end <= newString.length) {
    collector.push(newString.slice(start, end))
    if (end === newString.length) {
      // newString = newString.substring(1)
      newString = newString.slice(1)
      end = 0
    }
    end++ 
  }
  return collector
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: ['a', 'ab', 'abc', 'b', 'bc', 'c']");
  console.log("=>", consecutiveSubstrings('abc'));

  console.log("");

  console.log("Expecting: ['a']");
  console.log("=>", consecutiveSubstrings('a'));
}

module.exports = consecutiveSubstrings;

// Please add your pseudocode to this file
// And a written explanation of your solution
